src/main/java/appeng/core/Registration.java

```diff
@@ -338,9 +327,9 @@ void postInit( final FMLPostInitializationEvent event )
 	{
 		this.registerSpatial( true );
 
-		final IAppEngApi api = AEApi.instance();
+		final Api api = Api.INSTANCE;
```

Comment:
I thought about using an auto-generated implementation that retrieves the backing of the fields via the game registry, this would make the initialization code much nicer and allow us to separate the two.

I'll look into JEI. If this is possible, I'd construct the FeatureFactory outside the API and just inject it as a dependency when the API is created, that way i wouldn't need to access the actual API implementation above to facilitate the initialization phases.